src/index.ts

```diff
@@ -1,923 +1,1065 @@
-		log.warn("Calling mock '%s' instead using client '%s'", args.instance_id, client.name );
-		log.silly("Mock input: %s", JSON.stringify(args,null,4) );
-		resp			= await mocks( args );
-	    }
-	    else {
-		log.silly("Calling Conductor method (%s) over client '%s' with input: %s", method, client.name, JSON.stringify(args,null,4) );
-		resp			= await client.call( method, args );
-
-		if ( method === "call" ) {
-		    if ( typeof resp !== "string" )
-			log.warn("Expected 'ZomeApiResult' to be 'string', not '%s'", typeof resp );
-		    else {
-			let resp_length	= resp.length;
-			resp		= JSON.parse(resp);
-			log.debug("Parsed 'ZomeApiResult' response (length %s) to typeof '%s'", resp_length, typeof resp );
-		    }
-		}
-	    }
-	} catch ( err ) {
-	    // -32700
-	    //     Parse errorInvalid JSON was received by the server. An error occurred on the server while parsing the JSON text.
-	    // -32600
-	    //     Invalid RequestThe JSON sent is not a valid Request object.
-	    // -32601
-	    //     Method not foundThe method does not exist / is not available.
-	    // -32602
-	    //     Invalid paramsInvalid method parameter(s).
-	    // -32603
-	    //     Internal errorInternal JSON-RPC error.
-	    // -32000 to -32099
-	    //     Server errorReserved for implementation-defined server-errors.
-	    if ( err.code === -32000 ) {
-		if ( err.data.includes("response from service is not success") ) {
-		    log.error("Failed during Conductor call because of a signing request error: %s", err.data );
-		    throw new HoloError("Failed to get signatures from Client");
-		}
-		else {
-		    log.fatal("Failed during Conductor call with RPC Internal Error: %s -> %s", err.message, err.data );
-		    throw new HoloError("Unknown -32000 Error: %s", JSON.stringify( err ));
-		}
-	    } else if ( err instanceof Error ) {
-		log.error("Failed during Conductor call with error: %s", String(err) );
-		throw new HoloError(String(err));
-	    } else {
-		log.fatal("Failed during Conductor call with unknown error: %s", err );
-		throw new HoloError("Unknown RPC Error: %s", JSON.stringify( err ));
-	    }
-	}
-
-	log.normal("Call returned successful response: typeof '%s'", typeof resp );
-	return resp;
-    }
+  // --------------------------------------------------------------------------------------------
 
-    // Service Logger Methods
+  // Conductor Call Handling
 
-    addPendingConfirmation ( res_log_hash, agent_id, hha_hash ) {
-	log.info("Add response (%s) to pending confirmations with Agent/HHA: %s/%s", res_log_hash, agent_id, hha_hash );
-	this.pending_confirms[ res_log_hash ] = {
-	    agent_id,
-	    hha_hash,
-	};
+  validHoloHashPrefix(holoHashStrPrefix) {
+    switch (holoHashStrPrefix) {
+      case "uhCAk": // agent
+        return true;
+      case "uhCkk": // header
+        return true;
+      case "uhCEk": // entry
+        return true;
+      case "uhC0k": // dna
+        return true;
```

Comment:
not blocking, but you can also say:

```suggestion
      case "uhCAk": // agent
      case "uhCkk": // header
      case "uhCEk": // entry
      case "uhC0k": // dna
        return true;
```

Which is terser and I think makes the meaning clearer